File: bindings/java/NvsRuntime.java
// NaveScript Java/JNI Binding
public class NvsRuntime {
    static {
        System.loadLibrary("nvs_runtime");
    }
    
    private long handle;
    
    public NvsRuntime() {
        this.handle = createRuntime();
    }
    
    public native long createRuntime();
    public native void run(String code);
}

File: bindings/swift/NvsRuntime.swift
// NaveScript Swift Binding
import Foundation

public class NvsRuntime {
    private var handle: UnsafeMutableRawPointer
    
    public init() {
        self.handle = nvs_create_runtime()
    }
    
    public func run(code: String) {
        nvs_run(self.handle, code)
    }
}
